DB/src/test/java/io/deephaven/db/v2/utils/TestKeyedArrayBackedMutableTable.java

```diff
@@ -278,10 +267,24 @@ private void waitForCompletion() throws InterruptedException, ExecutionException
         }
     }
 
-    private void handleDelayedRefresh(BaseArrayBackedMutableTable kabut) {
-        new Thread(() -> {
-            SleepUtil.sleep(200);
-            LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(kabut::refresh);
-        }).start();
+    private void handleDelayedRefresh(final BaseArrayBackedMutableTable table, final FunctionalInterfaces.ThrowingRunnable<IOException> action) throws Exception {
+        final Thread refreshThread;
+        final Semaphore gate = new Semaphore(0);
+        table.setOnPendingChange(gate::release);
+        try {
+            refreshThread = new Thread(() -> {
+                gate.acquireUninterruptibly();
+                LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(table::refresh);
+            });
+            refreshThread.start();
+            action.run();
+
+        } finally {
+            table.setOnPendingChange(null);
```

Comment:
It seems like there may be another race here? We need to make sure that the LTM cycle has actually picked up our action - ie, I'm assuming that `action.run()`, invoking `mutableInputTable.add` is completed off thread? In which case, it's possible that we re-null-out setOnPendingChange before a cycle starts.

Reply:
No. Action.run() runs onPendingChange on the current thread.